lib/rules/object-shorthand.js

```diff
@@ -156,7 +156,7 @@ module.exports = {
                 property.value && property.value.id && property.value.id.name === property.key.name ||
 
                 // A property
-                property.value && property.value.name === property.key.name
+                property.value && (property.value.name || property.value.value) === (property.key.name || property.key.value)
```

Comment:
This comparison seems still wrong about computed properties and string literals.
Could you add tests such below?

- `({foo: "foo"})`
- `({"❤": "❤"})`
- `({[foo]: foo})`
- `({[foo]: "foo"})`

I guess `isRedudant` function should be:

```js
const astUtils = require("../ast-utils");

function isRedudant(property) {
    const value = property.value;

    if (value.type === "FunctionExpression") {
        return !value.id; // Only anonymous should be shorthand method.
    }
    if (value.type === "Identifier") {
        return astUtils.getStaticPropertyName(property) === value.name;
    }

    return false;
}
```